common/views/components/CopyUrl/CopyUrl.js

```diff
@@ -1,32 +1,103 @@
 // @flow
 
 import {spacing, font} from '../../../utils/classnames';
+import {Component} from 'react';
 import Icon from '../Icon/Icon';
 import HTMLInput from '../HTMLInput/HTMLInput';
 
 type Props = {|
   id: string,
-  url: string,
+  url: string
 |}
 
-const CopyUrl = ({id, url}: Props) => (
-  <div>
-    <HTMLInput
-      id='share'
-      type='text'
-      label='share url'
-      defaultValue={url}
-      isLabelHidden={true}
-      fontStyles={{s: 'HNL5', m: 'HNL4'}} />
-
-    <button aria-live='polite'
-      data-copy-text={url}
-      data-track-event={`{"category": "component", "action": "copy-url:click", "label": "id:${id}"}`}
-      className={`${spacing({s: 2}, {margin: ['top']})} ${font({s: 'HNM5', m: 'HNM4'})} btn btn--light is-hidden js-copy-url pointer`}>
-      <Icon name='check' extraClasses='icon--black is-hidden' />
-      <span className='js-copy-text'>Copy link</span>
-    </button>
-  </div>
-);
+type State = {|
+  isEnhanced: boolean,
+  isTextCopied: boolean,
+  buttonText: string
+|}
+
+class CopyUrl extends Component<Props, State> {
+  textInput: ?HTMLInputElement;
+  setTextInputRef: Function;
+  focusTextInput: Function;
+
+  constructor(props: Props) {
+    super(props);
+    this.textInput = null;
+
+    // Using refs to get access to elements for e.g. focusing
+    // https://reactjs.org/docs/refs-and-the-dom.html#callback-refs
```

Comment:
Why not use one of the, what seems to be, more standard ways?
Not saying we shouldn't do this - just trying to figure out what this way.

Reply:
For focusing an input, I really think this _is_ the most standard way. But it did feel pretty convoluted – if you know of any other/better ways, I'd be happy to implement.

Reply:
I think `refs` are a bit convoluted.

> Avoid using refs for anything that can be done declaratively.

> Don’t Overuse Refs
Your first inclination may be to use refs to “make things happen” in your app. If this is the case, take a moment and think more critically about where state should be owned in the component hierarchy. Often, it becomes clear that the proper place to “own” that state is at a higher level in the hierarchy. See the Lifting State Up guide for examples of this.

Reply:
Ain't no denying refs are a bit convoluted. I can't see myself ever wanting to _over_use them!